packages/rum-core/src/domain/rumEventsCollection/view/trackViewMetrics.ts

```diff
@@ -108,15 +108,15 @@ function trackActivityLoadingTime(
   callback: (loadingTimeValue: Duration | undefined) => void
 ) {
   const startTime = timeStampNow()
-  const { stop: stopWaitIdlePageActivity } = waitIdlePageActivity(lifeCycle, domMutationObservable, (params) => {
+  const subscription = createIdlePageActivityObservable(lifeCycle, domMutationObservable).subscribe((params) => {
     if (params.hadActivity) {
       callback(elapsed(startTime, params.endClocks.timeStamp))
     } else {
       callback(undefined)
     }
   })
```

Comment:
it seems a bit weird to me to use an observable when we expect a single notification.

Reply:
Yes, valid argument. I'll change it back.

Reply:
Sure, but IMO it:
- is using a more complex concept for no reason, we could use Promises even for synchronous API 
- can be misleading since I would expect that `IdlePageActivityObservable` notify us for every new page idle which is not the case

Reply:
> Sure, but IMO it:
> 
> * is using a more complex concept for no reason, we could use Promises even for synchronous API

Few reasons:

* We can't use Promise since it is not supported by IE11
* Promise cannot be aborted/stopped
* While I would agree that Observables like Rx are more complex than promises, our observable implementation is less complex than Promises.

> 
> * can be misleading since I would expect that `IdlePageActivityObservable` notify us for every new page idle which is not the case

Then it may be a naming issue? It could be renamed as `FirstIdlePageActivityObservable` or `NextIdlePageActivityObservable`.

Reply:
I was not proposing to use Promises here, it was just another example of using a more complex concept when it is not needed.

Reply:
I'm not sure to understand the complexity argument since the observable pattern is widely used in our code base. It means that developers already are use to it. However I do understand the misleading aspect of  `IdlePageActivityObservable`, it sure can be solved by changing the name but I'm fine with the callback.

Reply:
About complexity, it is not that observable is complex to use it is just that a callback would do the job.
Why involving a higher concept when a simpler one is enough?

Reply:
If it's for readability and consistency it's challengeable, but like I said I fine with the callback.

Reply:
I'm not sure it's worth more debate here, but: I had the exact same thought when Promises appeared in the JS ecosystem. We previously used callbacks (continuation passing style) or events everywhere, I didn't see the point of Promises at all and it took me a while to understand it and embrace it.

I'm not saying that it's exactly the same case for Observables. It hasn't reach the same consensus in JS ecosystem as Promises, and anyway the topic is highly subjective. But in my point of view, Observables provide the same advantages over callbacks than Promises. It helps to model and compose asynchronous scenarios, and could be used to unify our "data produced by a function over time that can be stopped" concept.

That said, I won't push for it. I'm happy to see more usages and to use it where it makes sense, but I'm not here to "evangelize". I think it's fine if we use both callbacks and observables.

Reply:
My driver in this debate is I'd like to prevent a [golden hammer](https://sourcemaking.com/antipatterns/golden-hammer) situation.
From my point of view, observables are suited to notify state changes to many dependents with a light coupling.
We are already using observables in cases where we don't need light coupling or don't have several dependents, I don't think we should actively change it. 
However, I'd prefer to not refactor existing code to use this pattern when the situation does not require it.

Reply:
> I'd prefer to not refactor existing code to use this pattern when the situation does not require it.

Isn't that the point of this PR?